src/models/hooks/objective.js

```diff
@@ -0,0 +1,195 @@
+// import { Op } from 'sequelize';
+import { auditLogger } from '../../logger';
+
+const autoPopulateObjectiveTemplateId = async (sequelize, instance, options) => {
+  // eslint-disable-next-line no-prototype-builtins
+  if (!instance.hasOwnProperty('objectiveTemplateId')
```

Comment:
So on the caller end do something link: autoPopulateObjectiveTemplateId({sequelize, instance, options});
Then on the implementation it would be autoPopulateObjectiveTemplateId(req), then use would be req.sequelize.

Reply:
I'm suggesting adding a function like:

```
function isDefined(object, prop) {
  return !object.hasOwnProperty(prop) || instance[prop] === null || instance[prop] === undefined
}
```

Then this line could be
```
if isDefined(instance, 'objectiveTemplateId') {
```

Reply:
because it seems there are some other places the new "isDefined" function could be used (might be a better name for it too!)